lib/Dashboard/TeamDashboardWidget.php

```diff
@@ -0,0 +1,160 @@
+use OCP\Dashboard\Model\WidgetItem;
+use OCP\Dashboard\Model\WidgetItems;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUserSession;
+use Psr\Log\LoggerInterface;
+
+class TeamDashboardWidget implements IAPIWidgetV2, IIconWidget, IButtonWidget, IConditionalWidget {
+	public function __construct(
+		private IURLGenerator $urlGenerator,
+		private IL10N $l10n,
+		private CircleService $circleService,
+		private ModelManager $modelManager,
+		private FederatedUserService $federatedUserService,
+		private ConfigService $configService,
+		private IUserSession $userSession,
+		private LoggerInterface $logger,
+	) {
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getItemsV2(string $userId, ?string $since = null, int $limit = 7): WidgetItems {
+		$circles = [];
+
+		try {
+			if (!$this->configService->getAppValueBool(ConfigService::FRONTEND_ENABLED)) {
+				throw new FrontendException('frontend disabled');
+			}
+
+			$user = $this->userSession->getUser();
+			$this->federatedUserService->setLocalCurrentUser($user);
+
+			$probe = new CircleProbe();
+			$probe->filterHiddenCircles()
+				->filterBackendCircles()
+				->setItemsLimit($limit)
+				->setItemsOffset($since ? (int)$since : 0);
+
+			$circles = array_map(function (Circle $circle) {
+				return new WidgetItem(
+					$circle->getDisplayName(),
+					'',
+					$this->urlGenerator->getAbsoluteURL($this->modelManager->generateLinkToCircle($circle->getSingleId())),
+					$this->urlGenerator->getAbsoluteURL($this->urlGenerator->linkToRoute('core.GuestAvatar.getAvatar', ['guestName' => $circle->getDisplayName(), 'size' => 64]))
+				);
+			}, $this->circleService->probeCircles($probe));
+		} catch (\Exception $e) {
+			$this->logger->error($e->getMessage(), ['exception' => $e]);
+		}
+		return new WidgetItems($circles);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getId(): string {
+		return 'circles';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getTitle(): string {
+		return 'Teams';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(): int {
+		return 0;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getIconClass(): string {
+		return 'icon-teams';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getUrl(): ?string {
+		return $this->getTeamPage();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function load(): void {
+	}
+
+	public function getWidgetButtons(string $userId): array {
+		return [
+			new WidgetButton(
+				WidgetButton::TYPE_MORE,
+				$this->getTeamPage(),
+				$this->l10n->t('Show all teams')
+			),
+			new WidgetButton(
+				WidgetButton::TYPE_SETUP,
+				$this->getTeamPage(),
+				$this->l10n->t('Create a new team')
+			),
+		];
+	}
+
+	public function getIconUrl(): string {
+		return $this->urlGenerator->getAbsoluteURL($this->urlGenerator->imagePath('circles', 'app.svg'));
+	}
+
+	private function getTeamPage(): string {
+		return $this->urlGenerator->getAbsoluteURL(
+			$this->urlGenerator->linkToRoute('contacts.page.index')
```

Comment:
this breaks when the contacts app is not enabled.
Circles is default enabled, but the contacts app is not even shipped:
https://github.com/nextcloud/server/blob/60681fb8240843b294db02593811a9c600b27335/core/shipped.json#L54

```
Technical details

    Request ID: 1WWtRvraU7IKjR7k0KyE
    Type: ReflectionException
    Code: -1
    Message: Class "OCA\Contacts\Controller\PageController" does not exist
    File: /var/www/html/lib/private/Route/Router.php
    Line: 469


Trace

#0 /var/www/html/lib/private/Route/Router.php(469): ReflectionClass->__construct()
#1 /var/www/html/lib/private/Route/Router.php(161): OC\Route\Router->getAttributeRoutes()
#2 /var/www/html/lib/private/Route/Router.php(400): OC\Route\Router->loadRoutes()
#3 /var/www/html/lib/private/Route/CachingRouter.php(65): OC\Route\Router->generate()
#4 /var/www/html/lib/private/URLGenerator.php(103): OC\Route\CachingRouter->generate()
#5 /var/www/html/apps/circles/lib/Dashboard/TeamDashboardWidget.php(153): OC\URLGenerator->linkToRoute()
#6 /var/www/html/apps/circles/lib/Dashboard/TeamDashboardWidget.php(123): OCA\Circles\Dashboard\TeamDashboardWidget->getTeamPage()
#7 /var/www/html/apps/dashboard/lib/Controller/DashboardController.php(78): OCA\Circles\Dashboard\TeamDashboardWidget->getUrl()
#8 [internal function]: OCA\Dashboard\Controller\DashboardController->OCA\Dashboard\Controller\{closure}()
#9 /var/www/html/apps/dashboard/lib/Controller/DashboardController.php(73): array_map()
#10 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(232): OCA\Dashboard\Controller\DashboardController->index()
#11 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(138): OC\AppFramework\Http\Dispatcher->executeController()
#12 /var/www/html/lib/private/AppFramework/App.php(184): OC\AppFramework\Http\Dispatcher->dispatch()
#13 /var/www/html/lib/private/Route/Router.php(338): OC\AppFramework\App::main()
#14 /var/www/html/lib/base.php(1065): OC\Route\Router->match()
#15 /var/www/html/index.php(49): OC::handleRequest()
#16 {main}
```

Maybe add that app enabled check to the `isEnabled` of the widget?